api/envoy/extensions/filters/udp/dns_filter/v3alpha/dns_filter.proto

```diff
@@ -52,8 +53,27 @@ message DnsFilterConfig {
     google.protobuf.Duration resolver_timeout = 1 [(validate.rules).duration = {gte {seconds: 1}}];
 
     // DNS resolution configuration which includes the underlying dns resolver addresses and options.
+    // *dns_resolution_config* will be deprecated once
+    // :ref:`typed_dns_resolver_config <envoy_v3_api_field_extensions.filters.udp.dns_filter.v3alpha.DnsFilterConfig.ClientContextConfig.typed_dns_resolver_config>`
+    // is fully supported.
```

Comment:
Rather than transition, did you consider incrementing the version to v4alpha?

Reply:
The consensus view of @envoyproxy/api-shepherds is that v4 is very far away. v3alpha is an acceptable place for breaking changes today, let's resolve the philosophical discussion on what alpha means in https://github.com/envoyproxy/envoy/issues/17920.